Dispatcher/Repository/ArticleRepository/ArticleRepository.swift

```diff
@@ -6,22 +6,18 @@
 //
 
 import Foundation
-class ArticleRepository: ArticleRepositoryProtocol  {    
-    let manager : NetworkManager = NetworkManager()
+
+class ArticleRepository: ArticleRepositoryProtocol {
+    let manager: NetworkManager = NetworkManager()
     
     func getArticles(completion: @escaping ([Card]) -> Void) {
         let apiKey = "ef5133dd364c41719494a74ce614d679"
         let articlesNumber = 7
-        let url = String("https://newsapi.org/v2/top-headlines?country=us&apiKey=\(apiKey)&pageSize=\(articlesNumber)")
-        manager.request(url: url, method: "get", completion: { data in
-            do {
-                let jsonData = try JSONDecoder().decode(NewsResponse.self, from: data as! Data)
-                let articles = jsonData.articles
-                print(articles)
-                completion(articles)
-            } catch {
-                print("Error decoding JSON: \(error.localizedDescription)")
-            }
-        })
+        let urlString = "https://newsapi.org/v2/top-headlines?country=us&apiKey=\(apiKey)&pageSize=\(articlesNumber)"
```

Comment:
please split this
the base url should be in the network manager - 
https://newsapi.org/v2
the apiKey should be in the netwrok manager - we do not want to duplicate it on every function
the articlesNumber should be a variable on the repository
the "endpoint" should be in the repository 
/top-headlines?country=us&apiKey=\(apiKey)&pageSize=\(articlesNumber)

Reply:
the apiKey could be static on the network manager 
and access it as 
NetworkManager.apiKey